include/igl/is_delaunay.cpp

```diff
@@ -0,0 +1,112 @@
+#include "is_delaunay.h"
+#include "unique_edge_map.h"
+#include <cassert>
+
+template <
+  typename DerivedV,
+  typename DerivedF,
+  typename DerivedD>
+IGL_INLINE void igl::is_delaunay(
+  const Eigen::MatrixBase<DerivedV> & V,
+  const Eigen::MatrixBase<DerivedF> & F,
+  Eigen::PlainObjectBase<DerivedD> & D)
+{
+  typedef Eigen::Matrix<typename DerivedF::Scalar,Eigen::Dynamic,2> MatrixX2I;
+  typedef Eigen::Matrix<typename DerivedF::Scalar,Eigen::Dynamic,1> VectorXI;
+  MatrixX2I E,uE;
+  VectorXI EMAP;
+  std::vector<std::vector<typename DerivedF::Scalar> > uE2E;
+  igl::unique_edge_map(F, E, uE, EMAP, uE2E);
+  const int num_faces = F.rows();
+  D.setConstant(F.rows(),F.cols(),false);
+  const auto D_at = [&D,&num_faces](const int he)->typename DerivedD::Scalar&
+  {
+    const int f = he%num_faces;
+    const int c = he/num_faces;
+    return D(f,c);
+  };
+  typedef typename DerivedV::Scalar Scalar;
+  // Should use Shewchuk's predicates instead.
+  const auto float_incircle = [](
```

Comment:
This is an inexact predicate?  Shewchuk's exact predicates is not easily templated, which does not fit the header-only convention.  What about Geogram's predicates?

Reply:
Geogram's predicate are also not header-only (but should be easier to isolate, see the [predicate psm](http://alice.loria.fr/software/geogram/doc/html/geogram_PSM.html)). But why not treat it as an external dependency that should be compiled as a pair of .cpp/.h? (Like we do for `stb_image` for example.)